Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/HotSpot.hs

```diff
@@ -0,0 +1,32 @@
+module Domain.Action.UI.HotSpot where
+
+import qualified Data.List as List
+import Domain.Types.HotSpot
+import Kernel.External.Maps
+import Kernel.Prelude
+import Kernel.Storage.Hedis (HedisFlow)
+import Kernel.Utils.Common
+import Storage.CachedQueries.CacheConfig
+import Storage.CachedQueries.Maps.LocationMapCache
+import Tools.Metrics
+
+frequencyUpdator ::
+  ( HasCacheConfig r,
+    CoreMetrics m,
+    HedisFlow m r,
+    HasFlowEnv m r '["hotSpotGeoHashPrecision" ::: Int]
+  ) =>
+  LatLong ->
+  m ()
+frequencyUpdator latLong = do
+  previousHotSpots <- getLocationsInCache
+  HotSpot {..} <- convertToHotSpot latLong
+  let mbTargetHotSpot = List.find (\hSpot -> hSpot.geoHash == geoHash) previousHotSpots
+  case mbTargetHotSpot of
+    Just foundHotSpot -> do
+      let otherHotSpot = List.filter (\hSpot -> hSpot.geoHash /= foundHotSpot.geoHash) previousHotSpots
+      let allHotSpots = HotSpot {frequency = foundHotSpot.frequency + 1, ..} : otherHotSpot
+      setLocationInCache allHotSpots
```

Comment:
lets use redis `HSET` for this ? [HINCRBY](https://redis.io/commands/hincrby) function would be better for this use case, instead of current way of `get -> filter -> caseMatch -> set` you can just update frequency for calculated geohash, if its not there it would be created else updated.

Reply:
didn't get you, please explain?